src/npc.cpp

```diff
@@ -1783,6 +1787,30 @@ Creature::Attitude npc::attitude_to( const Creature &other ) const
     return A_NEUTRAL;
 }
 
+void npc::npc_dismount()
+{
+    tripoint pnt;
```

Comment:
```suggestion
    cata::optional<tripoint> pnt;
```

And after the loop check for it being valid, and it not, return from the function:
```C++
if( !pnt ) {
    return;
}
... remaining cod to unmount.
```